pkg/controller/certificaterequests/util/reporter.go

```diff
@@ -56,6 +56,12 @@ func (r *Reporter) Failed(cr *cmapi.CertificateRequest, err error, reason, messa
 	r.recorder.Event(cr, corev1.EventTypeWarning, reason, message)
 	apiutil.SetCertificateRequestCondition(cr, cmapi.CertificateRequestConditionReady,
 		cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, message)
+
+}
+
+func (r *Reporter) InvalidRequest(cr *cmapi.CertificateRequest) {
```

Comment:
We need to also report information as to *why* the request failed (i.e. the 'reason' field) - if this method isn't used yet, it may be best to remove for now. Otherwise can you update the signature to ensure it's useful in future? 😄

Reply:
The idea is that others will be able to import and use this reporter so would like something here. I'll add a reason and message

Reply:
Sure - I'm just conscious of us breaking this method in future if someone else uses it, so we should be sure to expose everything we expect them to set now 😄